src/main/java/ch/wisv/connect/services/CHUserDetailsService.java

```diff
@@ -82,29 +82,15 @@ public CHUserDetails loadUserByGoogleCredential(SAMLCredential samlCredential) t
         String email = samlCredential.getNameID().getValue();
         Preconditions.checkArgument(StringUtils.isNotBlank(email), "email must not be blank");
 
-        // The username is the email address without the domain
-        String username = email.split("@")[0];
+        // The username is the email address without "@ch.tudelft.nl"
+        String username = email.replace("@ch.tudelft.nl", "");
 
         String meta = "email=" + email + ", username=" + username;
         log.debug("Loading user by {}", meta);
         // Get the person from Dienst2
         Person person = verifyMembership(dienst2Repository.getPersonFromGoogleUsername(username), meta);
 
-        Set<String> groups = new HashSet<>();
-        String[] rawGroups = samlCredential.getAttributeAsStringArray("groups");
```

Comment:
should we remove this?

Reply:
This piece of code parses the groups that are returned from the SAML response directly. but all these groups have to be linked manually to the SAML app in Google Admin. So this does not update when new groups are added. I thought it was better to remove it here, since we will now retrieve the groups via Dienst2

Reply:
Ah oke that makes sense!